### PR TITLE
Add "join table" for visits

### PR DESCRIPTION
**Pull request by SimonGideon**
## Project requirements

- Create a table named vets with the following columns:
- id: integer (set it as autoincremented PRIMARY KEY)
- name: string
- age: integer
- date_of_graduation: date
- There is a many-to-many relationship between the tables species and vets: a vet can specialize in multiple species, and a species can have multiple vets specialized in it. Create a "join table" called specializations to handle this relationship.
- There is a many-to-many relationship between the tables animals and vets: an animal can visit multiple vets and one vet can be visited by multiple animals. Create a "join table" called visits to handle this relationship, it should also keep track of the date of the visit.
- Insert the following data for vets:
- Vet William Tatcher is 45 years old and graduated Apr 23rd, 2000.
- Vet Maisy Smith is 26 years old and graduated Jan 17th, 2019.
- Vet Stephanie Mendez is 64 years old and graduated May 4th, 1981.
- Vet Jack Harkness is 38 years old and graduated Jun 8th, 2008.
- Insert the following data for specialties:
- Vet William Tatcher is specialized in Pokemon.
- Vet Stephanie Mendez is specialized in Digimon and Pokemon.
- Vet Jack Harkness is specialized in Digimon.
- Insert the following data for visits:
- Agumon visited William Tatcher on May 24th, 2020.
- Agumon visited Stephanie Mendez on Jul 22th, 2020.
- Gabumon visited Jack Harkness on Feb 2nd, 2021.
- Pikachu visited Maisy Smith on Jan 5th, 2020.
- Pikachu visited Maisy Smith on Mar 8th, 2020.
- Pikachu visited Maisy Smith on May 14th, 2020.
- Devimon visited Stephanie Mendez on May 4th, 2021.
- Charmander visited Jack Harkness on Feb 24th, 2021.
- Plantmon visited Maisy Smith on Dec 21st, 2019.
- Plantmon visited William Tatcher on Aug 10th, 2020.
- Plantmon visited Maisy Smith on Apr 7th, 2021.
- Squirtle visited Stephanie Mendez on Sep 29th, 2019.
- Angemon visited Jack Harkness on Oct 3rd, 2020.
- Angemon visited Jack Harkness on Nov 4th, 2020.
- Boarmon visited Maisy Smith on Jan 24th, 2019.
- Boarmon visited Maisy Smith on May 15th, 2019.
- Boarmon visited Maisy Smith on Feb 27th, 2020.
- Boarmon visited Maisy Smith on Aug 3rd, 2020.
- Blossom visited Stephanie Mendez on May 24th, 2020.
- Blossom visited William Tatcher on Jan 11th, 2021.
- Write queries to answer the following:
- Who was the last animal seen by William Tatcher?
- How many different animals did Stephanie Mendez see?
- List all vets and their specialties, including vets with no specialties.
- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
- What animal has the most visits to vets?
- Who was Maisy Smith's first visit?
- Details for most recent visit: animal information, vet information, and date of visit.
- How many visits were with a vet that did not specialize in that animal's species?
- What specialty should Maisy Smith consider getting? Look for the species she gets the most.
- Take a screenshot of the results of your queries.
- Remember to add the screenshot with the results of your queries to your Pull Request description.

Screenshot from 2023-02-04 07-35-14
Screenshot from 2023-02-04 07-31-51
Screenshot from 2023-02-04 07-20-17
Screenshot from 2023-02-04 07-05-23
Screenshot from 2023-02-04 07-04-58
Screenshot from 2023-02-04 07-03-44

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -29,5 +29,61 @@ UPDATE animals SET owner_id =3 WHERE name='Devimon' OR name='Plantmon';
...
+INSERT INTO visits (animal_id, vet_id, visit_date)
+SELECT a.id, v.id, '2020-05-14'
+FROM animals a, vets v
+WHERE a.name = 'Pikachu' AND v.name = 'Maisy Smith';
+
+INSERT INTO visits (animal_id, vet_id, visit_date)
+SELECT a.id, v.id, '2021-05-04'
+FROM animals a, vets v
+WHERE a.name = 'Devimon' AND v.name = 'Stephanie Mendez';
+
```
Comment (AnselemOdims) [inline data.sql:89]: - [ ] Good job working on the insert queries, but it is not complete with the project requirements. Kindly check below for the queries that are missing

image

File: queries.sql [hunk 2]
```diff
@@ -69,7 +69,47 @@ SELECT id, name,full_name, escape_attempts FROM animals INNER JOIN owners ON ani
...
+JOIN visits v ON a.id = v.animal_id
+JOIN vets ve ON v.vet_id = ve.id
+WHERE ve.name = 'Stephanie Mendez' AND v.visit_date BETWEEN '2020-04-01' AND '2020-08-30';
+
+SELECT a.name, COUNT(v.animal_id) AS visits
+FROM animals a
+JOIN visits v ON a.id = v.animal_id
+GROUP BY a.name
+ORDER BY visits DESC
+LIMIT 1;
```
Comment (AnselemOdims) [inline queries.sql:103]: - [ ] Good job working on these queries, but please kindly include queries for the below requirements.
  - Details for most recent visit: animal information, vet information, and date of visit.
  - How many times visits where with a vet that did not specialize in that animal's species?
  - What specialty should Maisy Smith consider getting? Look for the species she gets the most.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @SimonGideon , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: data.sql (hunk 1 above)
Comment (SimonGideon) [reply data.sql:89]: Noted. The concept is outlined:slightly_smiling_face:

**Review by Deepakdanger — APPROVED** [review]
Hi @SimonGideon , 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**